google-sheets/assets/index.html

```diff
@@ -0,0 +1,133 @@
+                        Live
+                    </aside>
+                </nav>
+            </header>
+        </div>
+        <main>
+            <div class="content">
+                <h1>
+                    <img src="https://twilio-labs.github.io/function-templates/static/v1/success.svg" />
+                    <div>
+                        <p>Welcome!</p>
+                        <p>Your live application with Twilio is ready to use!</p>
+                    </div>
+                </h1>
+                <section>
+                    <h2>Get started with your application</h2>
+                    <p>
+                        Now that your code is deployed, here are the last steps you need to do to finish your app.
+                    </p>
+                    <p>
+                        This app automatically logs texts to your phone number into the specified
+                        Google Sheets document.
+                    </p>
+                    <ol class="steps">
+                        <li>
+                            Share your Google Sheets document with the email address of your Google Sheets
+                            service account.
+                        </li>
+                        <li>Text your Twilio phone number with a message.</li>
+                        <li>
+                            Your spreadsheet should now contain the SID, phone number, and content of
+                            the text you sent.
+                        </li>
+                    </ol>
+                </section>
+
+                <section>
+                    <!-- APP_INFO_V2 -->
+                </section>
+                <section>
+                    <h2>Troubleshooting</h2>
+                    <ul>
+                        <p>
+                            <li>
+                                Verify your Google Sheets configuration:
+                                <div id="check-status">
+                                    <button onclick="showGoogleSheetsStatus()">Verify Configuration</button>
+                                    <span id="status-message"></span>
+                                </div>
+                                If the above check reveals problems, edit this function's <code>.env</code> file via the <code>Edit This Code</code> button above, if available, and ensure that all values are correct.
+                            </li>
+                        </p>
+                        <p>
+                            If you see "Google Sheets integration error: The caller does not have permission" in your
```

Comment:
I'm not sure i fully understand the failure case here, This seems to be the reason for the first step above. But in the scenario where its someone's personal google account that they used for the key generation, they still need to also register their email address for it?

Reply:
The keys are actually not for your own email but for a service account Google generates for you with its own email.

Reply:
That's correct, the email referenced here is the Google-generated service account email mentioned in the Getting Started steps. This very generic-sounding error is caused by the service account [not having permission to view or edit the Sheets document](https://stackoverflow.com/questions/38949318/google-sheets-api-returns-the-caller-does-not-have-permission-when-using-serve), and it seems like the vast majority of developers solve that by sharing the document with the service account.